### PR TITLE
QueryBuilder: tuning autoJoinOrderBy

### PR DESCRIPTION
**Pull request by EaredSeal**

### CODE REVIEW

File: src/Kdyby/Doctrine/QueryBuilder.php [hunk 1]
```diff
@@ -107,15 +107,27 @@ public function autoJoinOrderBy($sort, $order = NULL)
...
-			$sort = $alias . '.' . $sort;
+			$reg = "#([a-zA-Z_. ]+)\)#";
+			if(preg_match($reg, $sort, $matches)) {
+				$sortMix = $sort;
+				$sort = $matches[1];
+				$alias = $this->autoJoin($sort, "leftJoin");
+				$hiddenAlias = $alias . $sort . count($this->getDQLPart("orderBy"));
+
+				$this->addSelect(preg_replace($reg, $alias . '.' . $sort . ")", $sortMix) . " as HIDDEN " . $hiddenAlias);
+				$this->addGroupBy(reset(($this->getRootAliases())) . ".id");
```
Comment (fprochazka) [inline src/Kdyby/Doctrine/QueryBuilder.php]: why the group?

Comment (fprochazka) [inline src/Kdyby/Doctrine/QueryBuilder.php]: Oh, I see now...

File: src/Kdyby/Doctrine/QueryBuilder.php [hunk 2]
```diff
@@ -107,15 +107,27 @@ public function autoJoinOrderBy($sort, $order = NULL)
 		}
 
 		if (is_string($sort)) {
-			$alias = $this->autoJoin($sort);
-			$sort = $alias . '.' . $sort;
+			$reg = "#([a-zA-Z_. ]+)\)#";
```
Comment (fprochazka) [inline src/Kdyby/Doctrine/QueryBuilder.php]: I think you should also add the `\(` to the beggining so the regexp at https://github.com/Kdyby/Doctrine/pull/206/files#diff-f6cc3ca901f9632af758a516a922ba63R117 can be nicer

Comment (EaredSeal) [inline src/Kdyby/Doctrine/QueryBuilder.php]: You're right. I will send repair.

**Comment by fprochazka** [comment]
Merged https://github.com/Kdyby/Doctrine/commit/4e56faec5fa4bfce23eac6516f2c35dd2d7592b7 thank you! :)

**Pull request closed without merging**